### ISSUE TITLE
(Feature) Table: add a rownumber column type

### ISSUE THREAD

**Issue by anranruye** [ant-design-blazor/ant-design-blazor#1476]
### Is your feature request related to a problem? Please describe.
first inspired by #1455, row number column is a common column type for a table.

### Describe the solution you'd like
In fact, current column has a reference to `RowData`, but this is not exposed to user to use it. We can decide whether or not 
to expose it for all column types later. Now we can create a special column type to do this job.

**Comment by MutatePat** [comment]
How would this be implemented in practice? Since the table can be effected by filters/sorters, we would need an indexing stage when we are loading the DataSource initially. Perhaps storing a Dictionary<MetaData, TItem> with MetaData containing RowNumber and perhaps a timestamp.

However this development would be entirely ant-design-blazor as the react docs don't mention anything like this

**Comment by ElderJames** [comment]
@MutatePat Yes, my suggestion is to add the number, Id, or whatever you need to the data source so that it doesn't depend on the `RowData` of the table.

**Comment by anranruye** [comment]
@MutatePat 
Here,  row number means a fixed index,  it is not related to the DataSource. It only depends on the current position of a row in the table. When filter/sort happens,  a data item may have different row numbers. So RowData is enough. 

For index related to the DataSource,  as @ElderJames said,  users should add Id properties to their data models.
